python/cuproj/cuproj/cuprojshim/CMakeLists.txt

```diff
@@ -94,36 +103,34 @@ include(../../../../cpp/cuproj/cmake/thirdparty/get_rmm.cmake)
 
 add_library(cuprojshim STATIC src/cuprojshim.cu)
 
-set_target_properties(cuprojshim
-    PROPERTIES BUILD_RPATH                         "\$ORIGIN"
-               INSTALL_RPATH                       "\$ORIGIN"
-               # set target compile options
-               CXX_STANDARD                        17
-               CXX_STANDARD_REQUIRED               ON
-               CUDA_STANDARD                       17
-               CUDA_STANDARD_REQUIRED              ON
-               POSITION_INDEPENDENT_CODE           ON
-               INTERFACE_POSITION_INDEPENDENT_CODE ON
-)
+set_target_properties(
+  cuprojshim
+  PROPERTIES BUILD_RPATH "\$ORIGIN"
+             INSTALL_RPATH "\$ORIGIN"
+             # set target compile options
+             CXX_STANDARD 17
+             CXX_STANDARD_REQUIRED ON
+             CUDA_STANDARD 17
+             CUDA_STANDARD_REQUIRED ON
+             POSITION_INDEPENDENT_CODE ON
+             INTERFACE_POSITION_INDEPENDENT_CODE ON)
 
 # Use `CUPROJ_*_FLAGS` here because we reuse the cuProj's ConfigureCUDA.cmake above
-target_compile_options(cuprojshim
-            PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_FLAGS}>"
-                    "$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_FLAGS}>"
-)
+target_compile_options(cuprojshim PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_FLAGS}>"
+                                          "$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_FLAGS}>")
 
 # Use `CUPROJ_*_DEFINITIONS` here because we reuse the cuProj's ConfigureCUDA.cmake above
-target_compile_definitions(cuprojshim
-            PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_DEFINITIONS}>"
-                   "$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_DEFINITIONS}>"
-)
+target_compile_definitions(
+  cuprojshim PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUPROJ_CXX_DEFINITIONS}>"
+                    "$<$<COMPILE_LANGUAGE:CUDA>:${CUPROJ_CUDA_DEFINITIONS}>")
 
 # Specify include paths for the current target and dependents
-target_include_directories(cuprojshim
-           PUBLIC      "$<BUILD_INTERFACE:${CUPROJ_SOURCE_DIR}/include>"
-           PUBLIC     "$<BUILD_INTERFACE:${CUPROJSHIM_SOURCE_DIR}/include>"
```

Comment:
This is the only manual changes in this PR... fixes this finding from `cmake-lint`:

```text
python/cuproj/cuproj/cuprojshim/CMakeLists.txt:140,02: [E1122] Duplicate keyword argument PUBLIC
```

My read of https://cmake.org/cmake/help/latest/command/target_include_directories.html is that these forms of https://cmake.org/cmake/help/latest/command/target_include_directories.html are equivalent, so this is just a style thing and I don't think it should affect behavior at all.